plugin.cpp

```diff
@@ -416,316 +416,334 @@ void flattenDatapoint(Datapoint *datapoint,  string datapointName, vector<Datapo
+					std::vector<Datapoint *> newDatapoints;
+					bool isDatapoint = false;
 
-						// Iterate over split assets datapoints
-						for (string dpName : splitAssetDPs)
-						{
-							for (auto it = dps.begin(); it != dps.end(); it++ )
-							{
-								if (dpName == (*it)->getName())
-								{
-									Datapoint *dp = new  Datapoint(**it);
-									newDatapoints.emplace_back(dp);
-									isDatapoint = true;
-								}
-							}
-						}
-						if(isDatapoint)
+					// Iterate over split assets datapoints
+					for (string dpName : splitAssetDPs)
+					{
+						for (auto it = dps.begin(); it != dps.end(); it++ )
 						{
-							// Add new asset to reading set and asset tracker
-							newReadings.emplace_back(new Reading(newAssetName, newDatapoints));
-							if (tracker)
+							if (dpName == (*it)->getName())
 							{
-								tracker->addAssetTrackingTuple(info->configCatName, newAssetName, string("Filter"));
+								Datapoint *dp = new  Datapoint(**it);
+								newDatapoints.emplace_back(dp);
+								isDatapoint = true;
 							}
 						}
 					}
-				}
-				else // Split key doesn't exist
-				{
-					// Iterate over the datapoints
-					for (auto it = dps.begin(); it != dps.end(); it++)
+					if(isDatapoint)
 					{
-						std::string newAssetName = (*elem)->getAssetName();
-						newAssetName = newAssetName + "_" + (*it)->getName();
-						Datapoint *dp = new  Datapoint(**it);
-
 						// Add new asset to reading set and asset tracker
-						newReadings.emplace_back(new Reading(newAssetName, dp));
+						newReadings.emplace_back(new Reading(newAssetName, newDatapoints));
 						if (tracker)
 						{
-							tracker->addAssetTrackingTuple(info->configCatName, newAssetName, string("Filter"));
+							tracker->addAssetTrackingTuple(configCatName, newAssetName, string("Filter"));
 						}
 					}
 				}
 			}
-		}
-		else if (assetAction->actn == action::SELECT)
-		{
-			Reading *newReading = new Reading(**elem); // copy original Reading object
-                        // Iterate over the datapoints and remove unwanted
-                        vector<Datapoint *>& dps = newReading->getReadingData();
-			vector<string> rmList;
-			for (auto& dp : dps)
-                        {
-				string name = dp->getName();
-				bool found = false;
-                                for (auto& datapoint : assetAction->select)
+			else // Split key doesn't exist
+			{
+				// Iterate over the datapoints
+				for (auto it = dps.begin(); it != dps.end(); it++)
 				{
-					if (datapoint.compare(name) == 0)
+					std::string newAssetName = reading->getAssetName();
+					newAssetName = newAssetName + "_" + (*it)->getName();
+					Datapoint *dp = new  Datapoint(**it);
+
+					// Add new asset to reading set and asset tracker
+					newReadings.emplace_back(new Reading(newAssetName, dp));
+					if (tracker)
 					{
-						found = true;
-						break;
+						tracker->addAssetTrackingTuple(configCatName, newAssetName, string("Filter"));
```

Comment:
After doing the split you should probably look for any rules defined for the newly created assets.